src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -463,15 +470,58 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
             }
 
             return new FilteredCompletionModel(
-                GetHighlightedList(matchResults), index, filters,
+                GetHighlightedList(matchResults, filterText, highlightMatchingSpans, completionHelper), index, filters,
                 hardSelect ? UpdateSelectionHint.Selected : UpdateSelectionHint.SoftSelected,
                 centerSelection: true,
-                uniqueItem: moreThanOneMatchWithSamePriority ? null : bestMatchResult.GetValueOrDefault().VSCompletionItem);
+                uniqueItem: moreThanOneMatchWithSamePriority ? null : bestMatchResult.GetValueOrDefault().EditorCompletionItem);
         }
 
-        private static ImmutableArray<CompletionItemWithHighlight> GetHighlightedList(List<MatchResult> matchResults)
-            => matchResults.SelectAsArray(matchResult =>
-            new CompletionItemWithHighlight(matchResult.VSCompletionItem, matchResult.HighlightedSpans));
+        private static ImmutableArray<CompletionItemWithHighlight> GetHighlightedList(
+            List<MatchResult<VSCompletionItem>> matchResults,
+            string filterText,
+            bool highlightMatchingPortions,
+            CompletionHelper completionHelper)
+        {
+            return matchResults.SelectAsArray(matchResult =>
+            {
+                var highlightedSpans = GetHighlightedSpans(matchResult, completionHelper, filterText, highlightMatchingPortions);
+                return new CompletionItemWithHighlight(matchResult.EditorCompletionItem, highlightedSpans);
+            });
+
+            static ImmutableArray<Span> GetHighlightedSpans(
+                MatchResult<VSCompletionItem> matchResult,
+                CompletionHelper completionHelper,
+                string filterText,
+                bool highlightMatchingPortions)
+            {
+                if (!highlightMatchingPortions)
```

Comment:
nit: to avoid two returns of the same value, consider inverting this and placing the following code inside hte inverted if.